includes/Views/DspaceBooks.php

```diff
@@ -219,120 +219,109 @@ private function displayLinks( $start_here, $search_term ) {
+					'derivatives' => 'y',
+				),
+				'by-nc-sa' => array(
+					'license'     => 'standard',
+					'commercial'  => 'n',
+					'derivatives' => 'sa',
+				),
+				'by-nc-nd' => array(
+					'license'     => 'standard',
+					'commercial'  => 'n',
+					'derivatives' => 'n',
+				),
+			);
+			// interpret string as an object
+			$xml = $dspace_array;
+
+			if ( $xml ) {
+				$license = $this->metadataToCsv( $xml, 'dc.rights.uri' );
+				$license = strtolower( $license );
+				$license = explode( "/", $license );
+				$license = $license[4];
+				$title   = $this->metadataToCsv( $xml, 'dc.title' );
+				$lang    = $this->metadataToCsv( $xml, 'dc.language' );;
 			}
-		}
 
-		$key = array_keys( $expected[ $license ] );
-		$val = array_values( $expected[ $license ] );
+			$key = array_keys( $expected[ $license ] );
+			$val = array_values( $expected[ $license ] );
 
-		// build the url
-		$url = $endpoint . $key[0] . "/" . $val[0] . "/get?" . $key[1] . "=" . $val[1] . "&" . $key[2] . "=" . $val[2] .
-		       "&creator=" . urlencode( $authors ) . "&title=" . urlencode( $title ) . "&locale=" . $lang;
+			// build the url
+			$url = $endpoint . $key[0] . "/" . $val[0] . "/get?" . $key[1] . "=" . $val[1] . "&" . $key[2] . "=" . $val[2] .
+			       "&creator=" . urlencode( $authors ) . "&title=" . urlencode( $title ) . "&locale=" . $lang;
 
-		// go and get it
-		$c = curl_init( $url );
-		curl_setopt( $c, CURLOPT_RETURNTRANSFER, true );
-		curl_setopt( $c, CURLOPT_TIMEOUT, 20 );
+			// go and get it
+			$c = curl_init( $url );
+			curl_setopt( $c, CURLOPT_RETURNTRANSFER, true );
+			curl_setopt( $c, CURLOPT_TIMEOUT, 20 );
 
-		$response = curl_exec( $c );
-		curl_close( $c );
+			$response = curl_exec( $c );
+			curl_close( $c );
 
-		// if server response is not ok, return semi-meaningful string
-		if ( false == $response ) {
-			return 'license information currently unavailable from https://api.creativecommons.org/rest/1.5/';
-		}
+			// if server response is not ok, return semi-meaningful string
+			if ( false == $response ) {
+				return 'license information currently unavailable from https://api.creativecommons.org/rest/1.5/';
+			}
 
-		// in case response is not xml/invalid xml
-		libxml_use_internal_errors( true );
+			// in case response is not xml/invalid xml
+			libxml_use_internal_errors( true );
 
-		$obj = simplexml_load_string( $response );
-		$xml = explode( "\n", $response );
+			$obj = simplexml_load_string( $response );
+			$xml = explode( "\n", $response );
 
-		// catch errors, give them to php log
-		if ( ! $obj ) {
-			$errors = libxml_get_errors(); //@TODO do something with errors
-			foreach ( $errors as $error ) {
-				$msg = $this->displayXmlError( $error, $xml );
-				\error_log( $msg, 0 );
+			// catch errors, give them to php log
+			if ( ! $obj ) {
+				$errors = libxml_get_errors(); //@TODO do something with errors
+				foreach ( $errors as $error ) {
+					$msg = $this->displayXmlError( $error, $xml );
```

Comment:
this will produce a call to undefined function, fatal error. pls include the function, or if you think it can be used as a utility function, move it there and point/update all references to it.